server/aws-lsp-identity/src/sso/cache/fileSystemSsoCache.ts

```diff
@@ -0,0 +1,107 @@
+import { getSSOTokenFilepath, getSSOTokenFromFile, SSOToken } from '@smithy/shared-ini-file-loader'
+import { SsoCache, SsoClientRegistration, ssoClientRegistrationDuckTyper, ssoTokenDuckTyper } from './ssoCache'
+import { readFile, writeFile } from 'fs/promises'
+import { AwsError, tryAsync } from '../../awsError'
+import { AwsErrorCodes } from '@aws/language-server-runtimes/protocol'
+
+// As this is a cache, we tend to swallow a lot of the errors as it is ok to just recreate the items each
+// time, though recreating the SSO token without a refresh token will be every hour.  This is a better
+// situation that failing to get an SSO token at all.
+
+export class FileSystemSsoCache implements SsoCache {
```

Comment:
When cache will be used in server, is there a possibility of conflict of any kind when 2 or more IDE instances are running and will read/write data to cache? E.g. can 2 instances constantly rewrite and refresh each other tokens?
Or `clientName` and `ssoSessionName` will be unique per IDE running instance, so this is not an issue?

Reply:
Yes, this was discussed last week.  This is an open issue to resolve, which may be at the cache level or maybe at a higher level.  It is not addressable at this time though.  It is a corner case that can be addressed before the first destination usage.

Reply:
Got it, thanks for explanation. No concerns since we're tracking this issue